### ISSUE TITLE
ci: GitHub workflows fail to build container-images for multiple architectures

### ISSUE THREAD

**Issue by nixpanic** [csi-addons/kubernetes-csi-addons#92]
#89 changed how architectures are passed to the GitHub workflows. It seems that #91 was built only on one platform, instead of linux/amd64, linux/arm64 and linux/arm/v7.

For some reason the `secrets.BUILD_PLATFORMS` does not seem to be set for GitHub Actions?

**Comment by nixpanic** [comment]
Maybe there is some [GitHub access limitation](https://docs.github.com/en/actions/security-guides/encrypted-secrets#accessing-your-secrets):

> You can use and read encrypted secrets in a workflow file if you have access to edit the file.

The user creating the PR, will not have access to edit the file directly. Creating a PR from a branch in the local repository (#90) did work. That branch might have "access to edit the file"?

**Comment by nixpanic** [comment]
The behaviour is intentional and [documented in a GitHub security guide](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#accessing-secrets):

> Workflows triggered using the pull_request event have read-only permissions and have no access to secrets.

We'll need to think of an other way to limit building for multiple platforms.

**Comment by Madhu-1** [comment]
Yes due to security reason its done, sorry i forgot about it :( do we need to revert back to the PR introduced this one?

**Comment by nixpanic** [comment]
> Yes due to security reason its done, sorry i forgot about it :( do we need to revert back to the PR introduced this one?

Reverting is probably not needed. The builds after merging build all platforms, so that is fine for now.

I'd like some way of configuring it in a single place, so that it is easy to adjust in forks of this repo. Just not sure what/where that configuration should be done.
